### PR TITLE
Documentation for customizing Mattermost

### PR DESCRIPTION
**Pull request by lfbrock**
Can probably close https://github.com/mattermost/docs/issues/1006 after merging. 

If so, we may want to update the link under re-branding here to point to this page https://docs.mattermost.com/overview/faq.html#how-can-i-create-an-open-source-derivative-work-of-mattermost

@jasonblais - we may also want to mention plugins as a way to customize without forking if there's documentation to link to

### CODE REVIEW

File: source/deployment/customize-mattermost.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+8. Restart your Mattermost server 
+
+In Mattermost Enterprise Edition, it is possible to customize certain parts of the webapp without forking using our :doc:`Custom Branding <../administration/branding>` settings. 
+
+Mattermost Server
+-----------------
+
+There are a few things that can be customized in the Mattermost server without forking. 
+
+1. Modify text in the Mattermost interface by modifying the ``en.json`` file
```
Comment (lfbrock) [inline source/deployment/customize-mattermost.rst]: these were taking from IT's old GitHub issue, we may want to check if any of these items need to be updated

File: source/deployment/customize-mattermost.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Mattermost Mobile Applications
+------------------------------
+
+The Mattermost mobile applications can be customized if you choose to build the apps yourself. 
+
+To brand the mobile apps: 
+
+1. Fork the `mattermost-mobile <https://github.com/mattermost/mattermost-mobile>`_ repository
+2. Replace the name, images, and any key text strings
+3. Compile the apps
```
Comment (lfbrock) [inline source/deployment/customize-mattermost.rst]: Do we have documentation somewhere for compiling the mobile apps?

File: source/deployment/customize-mattermost.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+Mattermost Desktop Applications
+-------------------------------
+
+The Mattermost desktop applications can be customized if you choose to build the apps yourself. 
+
+To brand the desktop apps: 
+
+1. Fork the `mattermost/desktop <https://github.com/mattermost/desktop>`_ repository
+2. Replace the name, images, and any key text strings
+3. Compile the apps
```
Comment (lfbrock) [inline source/deployment/customize-mattermost.rst]: @jasonblais not sure if there's documentation we can link to about compiling desktop app, or anything else about desktop app deployment that would be useful to link here

Comment (jasonblais) [reply source/deployment/customize-mattermost.rst]: We could link to https://github.com/mattermost/desktop/blob/master/docs/development.md

File: source/deployment/customize-mattermost.rst (hunk 2 above)
Comment (jasonblais) [reply source/deployment/customize-mattermost.rst]: Amy is actually helping with this based off of Elias' drafts https://github.com/mattermost/docs/pull/1636

**Comment by jasonblais** [comment]
Re: plugins, we have docs, just not published yet.

It's currently hosted [here](http://mattermost-developer-documentation.s3-website-us-east-1.amazonaws.com/branches/master/extend/plugins/), just need the following before we can publish at developers.mattermost.com (tentatively next week)
 - add intro pages for webapp and server plugins
 - apply webpage styling we got from Asaad/Charles
 - get approved by marketing

**Comment by jasonblais** [comment]
@lindy65 Can you help ask a developer to review if the things to customize the Mattermost server are still accurate? See note: https://github.com/mattermost/docs/pull/1652#discussion_r152865173

File: source/deployment/customize-mattermost.rst (hunk 2 above)
Comment (jasonblais) [reply source/deployment/customize-mattermost.rst]: @lindy65 can you help add a link to https://docs.mattermost.com/mobile/mobile-compile-yourself.html?

File: source/deployment/customize-mattermost.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+In Mattermost Enterprise Edition, it is possible to customize certain parts of the webapp without forking using our :doc:`Custom Branding <../administration/branding>` settings. 
+
+Mattermost Server
+-----------------
+
+There are a few things that can be customized in the Mattermost server without forking. 
+
+1. Modify text in the Mattermost interface by modifying the ``en.json`` file 
+2. Replace the logo in email notifications by changing the file located in the ``/images`` directory
```
Comment (hmhealey) [inline source/deployment/customize-mattermost.rst]: These images are actually in the web app in its `/images` directory.

**Pull request merged**